### DISCUSSION TITLE
Storage requirement of livesync

### DISCUSSION THREAD

**Discussion by sh-l** [vrtmrz/obsidian-livesync#234] [Q&A]
I've been using apps like FolderSync on Android and iSH on iOS to manually sync vaults, and now considering migrating to this awesome plugin.

Before I actually start migration, I'd like to know a rough estimate on how much it would cost. Most of services listed in documentation seems to have free tier up to ~3GB-ish, so I'll need to pay for the storage in order to have this plugin running.

I have ~10GB of vault, which mainly consists of PDF files of academic papers.
I use MD files to provide links to PDF files. I often edit MD files, and also edit PDF files by adding some annotations to them via a separate PDF reading app.

In such a case, how would it be the estimated storage requirement for choosing a cloud DB service? Does it use some compression to reduce the size of the DB? Does it keep older versions of files, making the required storage size greater than the total file size?

Thank you.

**Comment by vrtmrz** [comment]
Thank you for asking me! And so thank you for the honour.

Unfortunately, the worst case costs 133-150% (non-encrypted) to 200-250% (encrypted) of all files. (If all chunks are unique).
However, As you mentioned, CouchDB may compress data. We would not be able to estimate the actual cost. We only can measure the actually used size by the local environment. We can install CouchDB on our desktop machine, and nothing else is needed.

Self-hosted LiveSync does not care about unreferenced chunks. So, sometimes we need to run garbage collection manually or rebuild the database.
And, 10GB may be a little too large to handle them in this plug-in. It might be better to skip synchronising stable-old PDFs (By folder and regular expression) and leave them to FolderSync.
